hedera-node/hedera-app/src/xtest/java/contract/WipeXTest.java

```diff
@@ -0,0 +1,176 @@
+                assertSuccess());
+    }
+
+    @Override
+    protected Map<ProtoBytes, AccountID> initialAliases() {
+        final var aliases = new HashMap<ProtoBytes, AccountID>();
+        aliases.put(ProtoBytes.newBuilder().value(OWNER_ADDRESS).build(), OWNER_ID);
+        return aliases;
+    }
+
+    @Override
+    protected Map<TokenID, Token> initialTokens() {
+        final var tokens = new HashMap<TokenID, Token>();
+        tokens.put(
+                ERC721_TOKEN_ID,
+                Token.newBuilder()
+                        .tokenId(ERC721_TOKEN_ID)
+                        .wipeKey(AN_ED25519_KEY)
+                        .totalSupply(1000)
+                        .treasuryAccountId(UNAUTHORIZED_SPENDER_ID)
+                        .tokenType(TokenType.NON_FUNGIBLE_UNIQUE)
+                        .build());
+        tokens.put(
+                ERC20_TOKEN_ID,
+                Token.newBuilder()
+                        .tokenId(ERC20_TOKEN_ID)
+                        .wipeKey(AN_ED25519_KEY)
+                        .totalSupply(1000)
+                        .treasuryAccountId(UNAUTHORIZED_SPENDER_ID)
+                        .tokenType(TokenType.FUNGIBLE_COMMON)
+                        .build());
+        return tokens;
+    }
+
+    @Override
+    protected Map<EntityIDPair, TokenRelation> initialTokenRelationships() {
+        final var tokenRelationships = new HashMap<EntityIDPair, TokenRelation>();
+        addErc721Relation(tokenRelationships, OWNER_ID, NUMBER_OWNED_NFTS);
+        addErc20Relation(tokenRelationships, OWNER_ID, 50L);
+        return tokenRelationships;
+    }
+
+    @Override
+    protected Map<NftID, Nft> initialNfts() {
+        final var nfts = new HashMap<NftID, Nft>();
+        nfts.put(
+                SN_1234,
+                Nft.newBuilder()
+                        .nftId(SN_1234)
+                        .ownerId(OWNER_ID)
+                        .spenderId(APPROVED_ID)
+                        .metadata(SN_1234_METADATA)
+                        .build());
+        return nfts;
+    }
+
+    @Override
+    protected Map<AccountID, Account> initialAccounts() {
+        final var accounts = new HashMap<AccountID, Account>();
+        accounts.put(
+                OWNER_ID,
+                Account.newBuilder()
+                        .accountId(OWNER_ID)
+                        .numberOwnedNfts(NUMBER_OWNED_NFTS)
+                        .alias(OWNER_ADDRESS)
+                        .build());
+        accounts.put(
+                UNAUTHORIZED_SPENDER_ID,
+                Account.newBuilder().accountId(UNAUTHORIZED_SPENDER_ID).build());
+        return accounts;
+    }
+
+    @Override
+    protected void assertExpectedAccounts(@NotNull final ReadableKVState<AccountID, Account> accounts) {
+        final var account = accounts.get(OWNER_ID);
+        assertNotNull(account);
+        assertEquals(NUMBER_OWNED_NFTS - 1, account.numberOwnedNfts());
```

Comment:
Could add an `assertExpectedTokenRelationships()` that the original balance of 50 fungible units decreased by 20.

Reply:
Yep, done. Thank you, seems I've missed that :)